numba/cuda/tests/cudapy/test_overload.py

```diff
@@ -1,7 +1,8 @@
-from numba import cuda, njit
-from numba.core.extending import overload
+from numba import cuda, njit, types
+from numba.core.errors import TypingError
+from numba.core.extending import overload, overload_attribute
 from numba.cuda.testing import CUDATestCase, skip_on_cudasim, unittest
-
+from numba.tests.test_extending import mydummy_type, MyDummyType
```

Comment:
Should the `make_dummy_type()` method on the `numba.tests.support.TestCase` be used to generate some new type so as to avoid cross test contamination?

Reply:
Yeah, I think that would work, and it does look a bit weird to use bits from `numba.tests.test_extending` here.

Reply:
Now changed.